pages/api/nft/index.ts

```diff
@@ -1,68 +1,106 @@
+    if(!networkName || proposalscMergeId < 0 || issueContractId < 0)
+      return res.status(400).json("Missing parameters");
+  
+    const settings = await models.settings.findAll({
     where: { 
       visibility: "public",
       group: "urls"
     },
     raw: true,
-  });
+    });
 
-  const defaultConfig = (new Settings(settings)).raw();
+    const defaultConfig = (new Settings(settings)).raw();
   
-  if (!defaultConfig?.urls?.ipfs)
-    return res.status(500).json("Missing ipfs url on settings");
-
-  const include = [
-    { association: "pullRequests" },
-    { association: "mergeProposals", where:{ contractId: +proposalscMergeId } },
-    { association: "repository" },
-  ];
-
-
-  const issue = await models.issue.findOne({
-    where: {
-      issueId,
-    },
-    include
-  });
-
-  if (!issue) return res.status(404).json('Issue not founded');
-
-  const proposal = issue.mergeProposals[0];
-  const pullRequest = issue.pullRequests.find((pr)=> pr.id === proposal.pullRequestId);
-  const user = await models.user.findOne({where: { githubLogin: pullRequest?.githubLogin }});
-
-  const nft = {
-    price: issue?.amount,
-    reward: rewardAmount,
-    githubHandle: user?.githubHandle,
-    repository: issue?.repository?.githubPath,
-    githubId: issue?.githubId,
-    githubPullRequestId: pullRequest.githubId
-  }
-  const { hash } = await ipfsService.add(nft, true);
+    if (!defaultConfig?.urls?.ipfs)
+      return res.status(500).json("Missing ipfs url on settings");
+
+    const customNetwork = await models.network.findOne({
+        where: {
+          name: {
+            [Op.iLike]: String(networkName)
+          }
+        }
+    });
+      
+    if(!customNetwork)
+      return res.status(404).json('Network not founded');
+
+    const DAOService = new DAO({ 
+      skipWindowAssignment: true,
+      web3Host: defaultConfig.urls.web3Provider,
+    });
+
+    if (!await DAOService.start()) return res.status(500).json("Failed to connect with chain");
+    if(!await DAOService.loadNetwork(customNetwork.networkAddress)) 
+      return res.status(500).json("network could not be loaded");
+
+    const network = DAOService.network;
+
+    await network.start();
+    
+    const networkBounty = await network.getBounty(issueContractId);
+    if (!networkBounty) return res.status(404).json("Invalid");
 
-  if(!hash) return res.status(500);
+    const proposal = networkBounty.proposals.find(p=> p.id === +proposalscMergeId)
 
-  const url = `${defaultConfig.urls.ipfs}/${hash}`;
+    if(!proposal)
+      return res.status(404).json("Invalid");
   
+    const participants = await Promise.all(proposal.details.map(async(detail: ProposalDetail) => {
+      if(!detail.recipient) return;
+
+      const user = await models.user.findOne({
+        where: { address: {
+        [Op.iLike]: String(detail.recipient)
+        } }});
+
+      const distributedAmount = networkBounty.tokenAmount * detail.percentage / 100;
```

Comment:
use the function [calculateDistributedAmounts](https://github.com/bepronetwork/web-network/blob/dev/helpers/calculateDistributedAmounts.ts) to get the real paid amount